lib/mutant/mutator/node/regexp.rb

```diff
@@ -0,0 +1,104 @@
+module Mutant
+  class Mutator
+    class Node
+
+      # Generic mutator
+      module RegularExpression
+        class Parser
+          def self.parse(source)
+            Node.new(Regexp::Parser.parse(source, 'ruby/2.3'))
+          end
+
+          class Node < DelegateClass(Regexp::Expression::Base)
```

Comment:
Instead to use an interface with mutable API, what about mapping the `Regexp::Expression::Base` to a `Parser::AST` node on custom `type` so we can use the same machinery, and reduce the need to even fight mutable state.